app/move/controllers/create/base.test.js

```diff
@@ -0,0 +1,127 @@
+          .calledOnce
+      })
+
+      it('should call check current location method', function() {
+        expect(controller.use).to.have.been.calledWith(
+          controller.checkCurrentLocation
+        )
+      })
+    })
+
+    describe('#middlewareLocals()', function() {
+      beforeEach(function() {
+        sinon.stub(FormController.prototype, 'middlewareLocals')
+        sinon.stub(controller, 'use')
+
+        controller.middlewareLocals()
+      })
+
+      it('should call parent method', function() {
+        expect(FormController.prototype.middlewareLocals).to.have.been
+          .calledOnce
+      })
+
+      it('should call set cancel url method', function() {
+        expect(controller.use).to.have.been.calledWith(controller.setCancelUrl)
+      })
+    })
+
+    describe('#setCancelUrl()', function() {
+      let res, nextSpy
+
+      beforeEach(function() {
+        nextSpy = sinon.spy()
+        res = {
+          locals: {},
+        }
+      })
+
+      context('with no moves url local', function() {
+        beforeEach(function() {
+          controller.setCancelUrl({}, res, nextSpy)
+        })
+
+        it('should set cancel url correctly', function() {
+          expect(res.locals).to.have.property('cancelUrl')
+          expect(res.locals.cancelUrl).to.equal(undefined)
+        })
+
+        it('should call next', function() {
+          expect(nextSpy).to.be.calledOnceWithExactly()
+        })
+      })
+
+      context('with moves url local', function() {
+        beforeEach(function() {
+          res.locals.MOVES_URL = '/moves?move-date=2019-10-10'
+          controller.setCancelUrl({}, res, nextSpy)
+        })
+
+        it('should set cancel url moves url', function() {
+          expect(res.locals).to.have.property('cancelUrl')
+          expect(res.locals.cancelUrl).to.equal('/moves?move-date=2019-10-10')
+        })
+
+        it('should call next', function() {
+          expect(nextSpy).to.be.calledOnceWithExactly()
+        })
+      })
+    })
+
+    describe('#checkCurrentLocation()', function() {
+      let req, nextSpy
+
+      beforeEach(function() {
+        nextSpy = sinon.spy()
+        req = {
+          session: {},
+        }
+      })
+
+      context('when current location exists', function() {
+        beforeEach(function() {
+          req.session = {
+            currentLocation: {},
+          }
+          controller.checkCurrentLocation(req, {}, nextSpy)
+        })
+
+        it('should call next without error', function() {
+          expect(nextSpy).to.be.calledOnceWithExactly()
+        })
+      })
+
+      context('when no current location exists', function() {
+        beforeEach(function() {
+          controller.checkCurrentLocation(req, {}, nextSpy)
+        })
+
+        it('should call next with error', function() {
+          expect(nextSpy).to.be.calledOnce
+          expect(nextSpy.args[0][0] instanceof Error).to.be.true
```

Comment:
more personal preference but you also have:
```
.to.be.an('error')
```

Reply:
Have amended this in #208 as well.